quadrat/inc/customization.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+require_once 'wp-customize-global-styles-setting.php';
+
+class GlobalStylesCustomizer {
+
+	private $custom_colors;
+
+	function __construct() {
+		add_action( 'customize_register', array( $this, 'set_customizations' ) );
+
+		add_action( 'customize_register', array( $this, 'register_section' ) );
+
+		/* Customizer Preview JS */
+		add_action( 'customize_preview_init', array( $this, 'customize_preview_js' ) );
+	}
+
+	/**
+	 * Gets the color from a CSS variable.
+	 *
+	 * This might already be in Gutenberg.
+	 */
+	function get_color( $color, $palette ) {
+		// Is this a HEX?
+		if ( 0 === strpos( $color, '#' ) ) {
+			// If so just return.
+			return $color;
+		}
+
+		// Is this a variable?
+		if ( 0 === strpos( $color, 'var(--wp--preset--color--' ) ) {
+			// If so just return the palette
+			$color_slug = preg_replace( '/var\(--wp--preset--color--(.*)\)/', '$1', $color );
+			$key        = array_search( $color_slug, array_column( $palette, 'slug' ), true );
+			return $palette[ $key ]['color'];
+		}
+	}
+
+	function set_customizations() {
+		$theme_json    = WP_Theme_JSON_Resolver::get_merged_data()->get_raw_data();
+		$color_palette = $theme_json['settings']['color']['palette'];
+
+		$this->custom_colors = array(
+			'name'        => __( 'Colors' ),
+			'type'        => 'section',
+			'slug'        => 'customize-global-styles',
+			'description' => __( 'Color Customization for Quadrat' ),
+			'controls'    => array(
+				array(
+					'name'     => __( 'Foreground Color' ),
+					'type'     => 'color',
+					'default'  => $this->get_color( $theme_json['styles']['color']['text'], $color_palette ),
```

Comment:
I just ran into an issue:

1. I change the foreground and background colors on this PR
2. Switched to a different PR, the color customizations persisted
3. Switched back to this PR to change the colors back to default
4. Clicking default in the color picker didn't work as expected — instead it just returned them to whatever custom color they were last set to.

Do you know why that would happen?